src/main/java/org/apache/maven/shared/dependency/analyzer/DefaultProjectDependencyAnalyzer.java

```diff
@@ -150,7 +155,7 @@ private static Set<Artifact> getTestArtifactsWithNonTestScope(Set<Artifact> test
         return nonTestScopeArtifacts;
     }
 
-    private Map<Artifact, Set<String>> buildArtifactClassMap(MavenProject project, ClassesPatterns excludedClasses)
+    protected Map<Artifact, Set<String>> buildArtifactClassMap(MavenProject project, ClassesPatterns excludedClasses)
```

Comment:
our internal dependency analyzer uses this method to compute used dependency classes which are provided by multiple artifacts